routes/dao_coin_exchange.go

```diff
@@ -0,0 +1,183 @@
+package routes
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/deso-protocol/core/lib"
+	"github.com/holiman/uint256"
+	"io"
+	"math/big"
+	"net/http"
+)
+
+type GetDAOCoinLimitOrdersRequest struct {
+	DAOCoin1CreatorPublicKeyBase58CheckOrUsername string `safeForLogging:"true"`
+	DAOCoin2CreatorPublicKeyBase58CheckOrUsername string `safeForLogging:"true"`
+}
+
+type GetDAOCoinLimitOrdersResponse struct {
+	Orders []DAOCoinLimitOrder
+}
+
+type DAOCoinLimitOrder struct {
+	TransactorPublicKeyBase58CheckOrUsername string `safeForLogging:"true"`
```

Comment:
do we ever actually return a username here? or is it only ever a base58 check pub key? If it's only ever a pub key, denote it as such. Same applies to the Buying / Selling attributes below.

I think it makes most sense to change these attributes to be Base58Check and include ProfileEntryResponses for each in order to provide username / other details needed.